### ISSUE TITLE
Settings overview (out of game)

### ISSUE THREAD

**Issue by Kruptein** [Kruptein/PlanarAlly#118]
The settings icon has been there for a while now, while there is no such as an account setting page yet.

Some things that this page should have include

- [ ] Account related
    - [ ] Change username
    - [x] Change password
    - [x] Add email address
    - [x] Change email address
    - [x] Remove email address
    - [x] Remove account
- [ ] Game related
    - [ ] Default settings for the various colors (ruler/grid/fog)

**Comment by Goury** [comment]
I'd very much like to see this application requiring an email address on registration, requiring administrative approval and activation of an account (and also to grant session creation permission) and also some way to restore a lost password using email. And also a way to lock registrations and allow administrators to make some accounts.

I'd like to propose these modes to be configurable via server config file.
I'd like to propose following role groups: administrators, DMs, players.
Administrators should be able to do stuff like role and access management.
DMs should be able to DM a session (not necessarily to create one, also limit the amount of sessions per DM would be nice).
Player should be able to move tokens and to do some stuff on a map (like draw, if allowed).
Account without a role should behave like a spectator.

Also it should be possible to grand someone a permission to grant permissions to others, like this leveled variants:
* Can grant others permission to do some stull
* Can grant others permission to grant this permission
* Can grant others permission to grant permission to grant this permission

Example: admin grants permission to DM a session to someone.
DM shouldn't be able make other DMs (unless he should).
DM should be allowed to grant players permission to draw on map.
Players who can draw shouldn't be able to grant the same permission to other players.

Tl; dr: **Serious access management** is what we need to make this software to the next level, role-and-permission based.

**Comment by Kruptein** [comment]
Hey just want to react, that I saw your comments and understand your desire/need for these features.

I've been a bit busy with my personal campaign and performance improvements to the lighting algorithm hence the slow progression on other tickets.  I'll try to squeeze in some more time in the coming months to tackle these outstanding tickets.

Thanks for the detailed explanation of what you would like btw!

**Comment by Goury** [comment]
@Kruptein =)
BTW, so far I failed to find development documentation, only user one and CONTRIBUTING.md file.
I'd love to join the development (I am also a bit busy nowdays and aiohttp and typescript are new to me) and I believe contributing to development documentation may be the best place to start for me.

**Comment by Kruptein** [comment]
Yes, one of the common pitfalls of hobby projects I'm afraid.

Working on "sexy" features is more fun than "boring" documentation, but the latter is nonetheless very important.

I've done some work in regards to user docs, although even that is still at a bare minimum right now. Development docs I've not really touched indeed.

**Comment by Goury** [comment]
How's about a wiki and a kanban board?

**Comment by Kruptein** [comment]
I already have a board right here on github, which I believe is publicly visible, although that might be wrong. (see the `projects` tab).  It's just one big board with things to do sorted according to what I think is important.  So it's not a proper sprint board or something alike.

A wiki is fine, we just need to think about what kind of wiki we would like to use.

**Comment by Goury** [comment]
Hmm. Github doesn't show me projects for this repo, but I can access https://github.com/Kruptein/PlanarAlly/projects/1 directly, but it's read-only for me.

As for wiki, I'd like to vote for dokuwiki, but any wiki should do.

**Comment by Kruptein** [comment]
I'm gonna be honest, I don't really like the "classic" style of wiki's for something like documentation.

I would rather go for something like [mkdocs with the material theme](https://squidfunk.github.io/mkdocs-material/) or something like readthedocs I currently have for user docs.  Both can easily be integrated into a PR flow as well.

**Comment by Kruptein** [comment]
Another one I came across today is [docute](https://docute.org/) which looks kinda nice.  I've not yet tried it, whereas I do have experience with mkdocs and readthedocs

**Comment by Kruptein** [comment]
in regards to the documentation which is actually kinda out of scope of this issue (so I would like it if any future discussion on this would get its own issue), I settled on gridsome.

You can see a preview (it's very empty so far) [over here](https://zealous-dubinsky-376acf.netlify.com/docs/)

Eventually I would also like to have a tab contributing next to docs and tutorial with info on how you can contribute to the project.

**Comment by Kruptein** [comment]
I'm gonna close this issue for now as the original issue is implemented in a branch currently.
I'll make a separate ticket for the permission/role stuff.
